### PR TITLE
Localization automation

### PR DESCRIPTION
**Pull request by adityamulik**
- Role created for extracting strings from a project, uploading the strings to Memsource Client for translation.
- Role created for extracting translated strings from memsource, pushing it back to the project repository by creating a pull request.
- Two playbooks each for it's respective pre-translation and post-translation has been created.
- Variables are defined in group vars.

### CODE REVIEW

File: roles/pre_translation/tasks/upload_strings_to_memsource.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+---
+
+# - name: Assertions
```
Comment (rooftopcellist) [inline roles/pre_translation/tasks/upload_strings_to_memsource.yml:3]: We should be able to un-comment this now.

**Review by rooftopcellist — CHANGES_REQUESTED** [review]

**Comment by rooftopcellist** [comment]
If I am understanding this correctly, this role will one work for a single project at a time.  So if a team had say, multiple repos that they wanted to push/pull strings from, they could achieve this with Controller by creating multiple job templates using the same playbook (for ex.: pre_translation.yml) for each repo, and string them together as a Workflow Job template.  Or they could achieve something similar with something like a Jenkins pipeline.

File: roles/pre_translation/tasks/upload_strings_to_memsource.yml (hunk 1 above)
Comment (adityamulik) [reply roles/pre_translation/tasks/upload_strings_to_memsource.yml:3]: Uncommented, tested and will open a PR for the same.

File: roles/pre_translation/.travis.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+---
+language: python
+python: "2.7"
```
[the remark this answers is not in the document]
Comment (adityamulik) [reply roles/pre_translation/.travis.yml:3]: Updated the version to 3.8 for both roles, will merge it with opening a PR

File: roles/pre_translation/tasks/extract_strings_from_upstream.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+---
+
+- name: Temporate Repository Storage Directory
```
[the remark this answers is not in the document]
Comment (adityamulik) [reply roles/pre_translation/tasks/extract_strings_from_upstream.yml:3]: Updated typos as per PR https://github.com/ansible/ansible-collection-memsource/pull/20

File: roles/pre_translation/.travis.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (adityamulik) [reply roles/pre_translation/.travis.yml:3]: Updated per PR https://github.com/ansible/ansible-collection-memsource/pull/20

File: roles/pre_translation/meta/main.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
+galaxy_info:
+  author: your name
+  description: your role description
+  company: your company (optional)
```
[the remark this answers is not in the document]
Comment (adityamulik) [reply roles/pre_translation/meta/main.yml:4]: Details have been filled as per https://github.com/ansible/ansible-collection-memsource/pull/20

File: roles/pre_translation/tasks/upload_strings_to_memsource.yml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+#   assert:
+#     that:
+#       - memsource_username | default('') | length
+#       - memsource_password | default('') | length
```
[the remark this answers is not in the document]
Comment (adityamulik) [reply roles/pre_translation/tasks/upload_strings_to_memsource.yml:7]: fixed per pr https://github.com/ansible/ansible-collection-memsource/pull/20

**Comment by adityamulik** [comment]
@rooftopcellist Yes, so each repository will have an individual project on Memsource for translation and similarly pull/ push roles will be applicable for each project.

**Comment by rooftopcellist** [comment]
@adityamulik technically, each _component_ (awx, galaxy, pinakes) will have it's own memsource project with api and ui strings pushed to it.  

> @rooftopcellist Yes, so each repository will have an individual project on Memsource for translation and similarly pull/ push roles will be applicable for each project.

**Comment by adityamulik** [comment]
This PR is good to be closed without merging

**Pull request closed without merging**